c/opentx.h

```diff
@@ -162,6 +162,48 @@ void hash_cache_append2(HashCache *cache, uint8_t *buf, size_t len) {
   blake2b_update(&cache->state, buf, len);
 }
 
+int hash_cell_script(HashCache *cache, size_t index, size_t source,
+                     size_t field, uint16_t cell_mask) {
+  uint8_t script[OPENTX_SCRIPT_SIZE];
+  uint64_t len = OPENTX_SCRIPT_SIZE;
+  int err =
+      ckb_checked_load_cell_by_field(script, &len, 0, index, source, field);
+  if (err != CKB_SUCCESS) {
```

Comment:
If it's CKB_INDEX_OUT_OF_BOUND, should return failed.
If it's CKB_ITEM_MISSING, it's only allowed for type script. Not allowed for lock script.